### ISSUE TITLE
GammaRay don't compile when glslang is installed and using -DCMAKE_DISABLE_FIND_PACKAGE_Glslang=TRUE to ignore it

### ISSUE THREAD

**Issue by gbschenkel** [KDAB/GammaRay#674]
### Versions
**Distro** Slackware 15.0 rc3 (up-to-date)
**CMake** 3.21.4
**GCC** 11.2.0
**Make** 4.3
**Qt** 15.3
**GammaRay** 2.11.3
**KDSME** currently not installed since was just recommended

### Problem
Hi, I having trouble compiling GammaRay from source, it appears only works building using one processing unit, `make -j1`, instead of using all available, in my case 29, `make -j$(expr $(nproc) + 1)`

Also using `make -j1` I got the following Issue:
```bash
[ 86%] Generating widget_wireframe.frag_widget_wireframe.geom_widget_wireframe.vert_validate_program_3.stamp
/home/schenkel/Development/gitrepo/slackbuilds/misc/gammaray/GammaRay-2.11.3/plugins/widgetinspector/assets/shaders/widget_wireframe.frag
/home/schenkel/Development/gitrepo/slackbuilds/misc/gammaray/GammaRay-2.11.3/plugins/widgetinspector/assets/shaders/widget_wireframe.geom
/home/schenkel/Development/gitrepo/slackbuilds/misc/gammaray/GammaRay-2.11.3/plugins/widgetinspector/assets/shaders/widget_wireframe.vert
ERROR: Linking vertex stage: Types must match:
    anon@0: " out block{ gl_Position 4-component vector of float Position gl_Position,  gl_PointSize float PointSize gl_PointSize,  out 1-element array of float ClipDistance gl_ClipDistance,  gl_ClipVertex 4-component vector of float ClipVertex gl_ClipVertex,  out 4-component vector of float FrontColor gl_FrontColor,  out 4-component vector of float BackColor gl_BackColor,  out 4-component vector of float FrontSecondaryColor gl_FrontSecondaryColor,  out 4-component vector of float BackSecondaryColor gl_BackSecondaryColor,  out 1-element array of 4-component vector of float TexCoord gl_TexCoord,  out float FogFragCoord gl_FogFragCoord}" versus gl_in: " in 3-element array of block{ in 4-component vector of float Position gl_Position,  in float PointSize gl_PointSize,  in 1-element array of float ClipDistance gl_ClipDistance}"

make[2]: *** [plugins/widgetinspector/assets/shaders/CMakeFiles/widget_wireframe.frag_widget_wireframe.geom_widget_wireframe.vert_validate_program_3.dir/build.make:75: plugins/widgetinspector/assets/shaders/widget_wireframe.frag_widget_wireframe.geom_widget_wireframe.vert_validate_program_3.stamp] Error 3
make[1]: *** [CMakeFiles/Makefile2:9658: plugins/widgetinspector/assets/shaders/CMakeFiles/widget_wireframe.frag_widget_wireframe.geom_widget_wireframe.vert_validate_program_3.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```
The whole command from source dir:
```bash
mkdir build
cd build
cmake -DCMAKE_C_FLAGS:STRING="-O2 -fPIC" -DCMAKE_CXX_FLAGS:STRING="-O2 -fPIC" -DCMAKE_INSTALL_PREFIX=/usr -DMAN_INSTALL_DIR=/usr/man -DCMAKE_BUILD_TYPE=Release ..
make -j1
```
I have tried with the master branch, same problem.
I will try compile KDSME to see if the problem still persist.

**Comment by gbschenkel** [comment]
Okay, just found out this is related to #635. On Slackware the `glslangValidator` is part of Vulkan-SDK.
```bash
root@saitama:~# slackpkg file-search glslangValidator

Looking for glslangValidator in package list. Please wait... DONE

The list below shows the packages that contains "glslangValidator" file.

[inst] multilib    : vulkan-sdk-compat32-1.2.176.1-x86_64-2compat32
[inst] slackware64 : vulkan-sdk-1.2.176.1-x86_64-2

You can search specific packages using "slackpkg search package".
```

**Comment by gbschenkel** [comment]
Okay, since the project don't have `-DGlslang=OFF`, I have edit the cmake/FindGlslang.cmake to never find Glslang.
```diff
diff -Naur cmake/FindGlslang.cmake-old cmake/FindGlslang.cmake
--- cmake/FindGlslang.cmake-old 2022-01-28 20:02:54.110431215 -0300
+++ cmake/FindGlslang.cmake     2022-01-28 23:43:18.781299683 -0300
@@ -27,9 +27,6 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 find_program(GLSLANG_VALIDATOR_EXECUTABLE glslangValidator)
-if (GLSLANG_VALIDATOR_EXECUTABLE)
-  set(Glslang_FOUND TRUE)
-endif()
 
 # Copied from ECMPoQmTools which copied it from FindGettext.cmake
 function(_glslang_get_unique_target_name _name _unique_name)
```
After disabling `glslang` I was able to build it with all processing units.

**Comment by winterz** [comment]
cmake -DCMAKE_DISABLE_FIND_PACKAGE_GLSANG

**Comment by gbschenkel** [comment]
> cmake -DCMAKE_DISABLE_FIND_PACKAGE_GLSANG

@winterz I have tried, this don't work.
```cmake
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_DISABLE_FIND_PACKAGE_GLSANG
```
I tried using `-DCMAKE_DISABLE_FIND_PACKAGE_GLSLANG=TRUE` and `-DCMAKE_DISABLE_FIND_PACKAGE_GLSLANG_VALIDATOR=TRUE`

I also tried `-DCMAKE_DISABLE_FIND_PACKAGE_Glslang=TRUE` which appears be the right option but:
```cmake
CMake Error at plugins/widgetinspector/assets/shaders/CMakeLists.txt:15 (glslang_validate_program):
  Unknown CMake command "glslang_validate_program".
```
Appears some work need be done on `cmake/FindGlslang.cmake`.

Please reopen this issue.

**Comment by winterz** [comment]
you're right:  `-DCMAKE_DISABLE_FIND_PACKAGE_Glslang=TRUE` is the correct invocation
reopening this one.

we probably need some if(Glslang_FOUND) conditionals

**Comment by winterz** [comment]
fixed in a4439b91928206c8572c8e43eed2498716b732a9

**Comment by gbschenkel** [comment]
Thanks, I gonna update the Slackware script with this patch until the next release
-- 
Enviado de meu dispositivo Android com K-9 mail. Desculpe-me pela brevidade.

Em 1 de fevereiro de 2022 19:30:32 BRT, Allen Winter ***@***.***> escreveu:
